econpizza/config.py

```diff
@@ -0,0 +1,75 @@
+"""Configuration object"""
+
+import os
+import jax
+
+class EconPizzaConfig(dict):
+    def __init__(self, *args, **kwargs):
+        super(EconPizzaConfig, self).__init__(*args, **kwargs)
+        self._enable_persistent_cache = False
+        self._econpizza_cache_folder = "__econpizza_cache__"
+        self._jax_cache_folder = "__jax_cache__"
+
+    @property
+    def enable_persistent_cache(self):
+        return self._enable_persistent_cache
+
+    @enable_persistent_cache.setter
+    def enable_persistent_cache(self, value):
+        self._enable_persistent_cache = value
+
+    @property
+    def jax_cache_folder(self):
+        return self._jax_cache_folder
+    
+    @jax_cache_folder.setter
+    def jax_cache_folder(self, value):
+        self._jax_cache_folder = value
+    
+    @property
+    def econpizza_cache_folder(self):
+        return self._econpizza_cache_folder
+    
+    @econpizza_cache_folder.setter
+    def econpizza_cache_folder(self, value):
+        self._econpizza_cache_folder = value
+
+    def update(self, key, value):
+        if hasattr(self, key):
+            setattr(self, key, value)
+        else:
+            raise AttributeError(f"'EconPizzaConfig' object has no attribute '{key}'")
+    
+    def _create_cache_dir(self, folder_name: str):
+        cwd = os.getcwd()
+        folder_path = os.path.join(cwd, folder_name)
+        os.makedirs(folder_path, exist_ok=True)
+
+        return folder_path
+
+    def setup_persistent_cache(self):
+        """Create folders for JAX and EconPizza cache.
+        By default, they are created in callee working directory.
+        """
+        if self.enable_persistent_cache == True:
+            folder_path_pizza = self._create_cache_dir(self.econpizza_cache_folder)
+            folder_path_jax = self._create_cache_dir(self.jax_cache_folder)
+
+            jax.config.update("jax_compilation_cache_dir", folder_path_jax)
```

Comment:
A potential issue here is that a user might already have set persistent cache for JAX. Arguably, this is not very likely, but if so, this may cause confusion. So the ep default path `__jax_cache__` should only be used if 'jax_compilation_cache_dir' is not yet set.

Reply:
Ah, great point!